src/core/stdcpp/allocator.d

```diff
@@ -283,8 +283,15 @@ version (CppRuntime_Microsoft)
         version = INTEL_ARCH;
 
     // HACK: should we guess _DEBUG for `debug` builds?
-    version (NDEBUG) {}
-    else debug version = _DEBUG;
+    version (_DEBUG)
+        enum _DEBUG = true;
+    else version (NDEBUG)
+        enum _DEBUG = false;
+    else
+    {
+        import core.stdcpp.xutility : __CXXLIB__;
+        enum _DEBUG = __CXXLIB__.length && 'd' == __CXXLIB__[$-1]; // libcmtd, msvcrtd
```

Comment:
I don't know if this is good. If the logic that drives the codegen doesn't match C++, then you'll get weird link errors that are hard for a user to diagnose.